pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -40,7 +40,7 @@ def work(self):
 
         if self.evolve_all[0] != 'all':
             # check for negation
-            negate = filter(lambda x: x[0] == '-', self.evolve_all)
+            negate = filter(lambda x: len(x) > 0 and x[0] == '-', self.evolve_all)
```

Comment:
you mean if `"evolve_all": "none"` or `"evolve_all": "Pidgey, none"` ?

The first will not get to this line as it will be return at 37.
The second one will not be in negate as it doesn't have -. Even if someone try to do "Pidgey, -none", it won't match any pokemon.